components/ias-adapter/internal/storage/postgres/tenant_mapping.go

```diff
@@ -0,0 +1,85 @@
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/kyma-incubator/compass/components/ias-adapter/internal/errors"
+	"github.com/kyma-incubator/compass/components/ias-adapter/internal/types"
+)
+
+const tenantMappingsTable = "tenant_mappings"
+
+var (
+	upsertTenantMappingQuery  = fmt.Sprintf("INSERT INTO %s (formation_id, ucl_application_id, value) VALUES ($1, $2, $3) ON CONFLICT ON CONSTRAINT pk DO UPDATE SET value=$3", tenantMappingsTable)
+	selectTenantMappingsQuery = fmt.Sprintf("SELECT value FROM %s WHERE formation_id=$1", tenantMappingsTable)
+	deleteTenantMappingQuery  = fmt.Sprintf("DELETE FROM %s WHERE formation_id=$1 AND ucl_application_id=$2", tenantMappingsTable)
+)
+
+func (c Connection) UpsertTenantMapping(ctx context.Context, tenantMapping types.TenantMapping) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, c.cfg.RequestTimeout)
```

Comment:
The timeout logic could be encapsulated in a struct implementing the `pgxPool` interface, could be the `Connection` struct itself (in private methods). In this way we will have a single place (still, several functions) in which we deal with the timeout (and with a quick look at it we can check that it is used in every db operation) and will not introduce additional lines here (which slightly distracts the reader from the main logic of the method).